crates/iceberg/src/spec/table_metadata.rs

```diff
@@ -461,9 +461,57 @@ impl TableMetadata {
         file_io: &FileIO,
         metadata_location: impl AsRef<str>,
     ) -> Result<()> {
+        use std::io::Write as _;
+
+        use flate2::write::GzEncoder;
+
+        let json_data = serde_json::to_vec(self)?;
+
+        // Check if compression is enabled via table properties
+        let codec = self
+            .properties
+            .get(crate::spec::table_properties::TableProperties::PROPERTY_METADATA_COMPRESSION_CODEC)
+            .map(|s| s.as_str())
+            .unwrap_or(crate::spec::table_properties::TableProperties::PROPERTY_METADATA_COMPRESSION_CODEC_DEFAULT);
+
+        let (data_to_write, actual_location) = match codec {
+            "gzip" => {
```

Comment:
nit: make this comparison case insensitive 

https://github.com/apache/iceberg/blob/700575f6e58c655ba680a0e5cc25d5a88c225d3c/core/src/main/java/org/apache/iceberg/TableMetadataParser.java#L63